guacamole-ext/src/main/java/org/apache/guacamole/token/TokenName.java

```diff
@@ -63,35 +57,38 @@
     /**
      * This utility class should not be instantiated.
      */
-    private TokenName() {}
+    protected TokenName() {}
 
     /**
      * Generates the name of the parameter token that should be populated with
-     * the value of the given LDAP attribute. The name of the LDAP attribute
-     * will automatically be transformed from "CamelCase", "headlessCamelCase",
+     * the value of the given attribute. The name of the attribute will
```

Comment:
While the LDAP version of this was tied to attributes, a generic version should probably not be.

Reply:
... and I guess that means the function should probably be renamed, too, with `fromAttribute()` being overly specific.

Reply:
So, I removed the LDAP stuff, but kept the `fromAttribute()` name - it seems like most of the extensions still call these attributes, so maybe it isn't too specific?  I'm open to suggestions, but seems okay to me...

Reply:
Well ... they call them attributes because that's what they happen to be in those cases: attributes on an LDAP object and attributes on a CAS principal. I don't think that will fit in the general case, where we are providing a `TokenName` class to serve as a convenience for extensions that wish to generate tokens dynamically. Those tokens may come from something called an attribute, perhaps even Guacamole's own attributes, or they may not.

There's a similar function on the JavaScript side for translation strings which we called `canonicalize()`:

https://github.com/apache/guacamole-client/blob/d1e928bea79ca81c827e9b6adedabc98eefdf701/guacamole/src/main/webapp/app/locale/services/translationStringService.js#L28-L42

Perhaps that would be a better choice?

Reply:
Sounds good to me.  Changed as requested.